main.js

```diff
@@ -75,15 +75,22 @@ async function run() {
     fs.writeFileSync(cmd, wrap);
 
     core.addPath(dest);
-    const c_paths = [(p_release ? dest : 'C:') + `\\msys64\\var\\cache\\pacman\\pkg\\`];
+    const pkgCachePath = (p_release ? dest : 'C:') + `\\msys64\\var\\cache\\pacman\\pkg\\`;
 
     core.exportVariable('MSYSTEM', p_msystem);
 
-    if (p_cache === 'true') {
-      core.startGroup('Restoring cache...');
-      console.log('Cache ID:', await cache.restoreCache(c_paths, 'msys2', ['msys2-']));
-      core.endGroup();
-    }
+    // We want a cache key that is ideally always the same for the same kind of job.
+    // So that mingw32 and ming64 jobs, and jobs with different install packages have different caches.
+    let shasum = crypto.createHash('sha1');
+    ['release', 'update', 'path-type', 'msystem', 'install'].map((n) => { return shasum.update(core.getInput(n)); });
```

Comment:
Since these are already retrieved, I think it'd be better to `[p_release, p_update, p_pathtype, p_msystem, p_install]`.

Reply:
done